validator/server_arb/machine.go

```diff
@@ -383,9 +394,16 @@ func (m *ArbitratorMachine) SetPreimageResolver(resolver GoPreimageResolver) err
 		return errors.New("machine frozen")
 	}
 	id := lastPreimageResolverId.Add(1)
```

Comment:
I moved back to represent contextId as a pointer, so it is more clear when a machine doesn't have a preimage resolver attached to it, and to avoid worrying about which numbers are valid IDs.